### DISCUSSION TITLE
Visualize the feature maps in Detectron2

### DISCUSSION THREAD

**Discussion by lyhsieh** [facebookresearch/detectron2#3268] [Q&A] [answered]
Hello everyone!
We know that Detectron2 uses ResNet as the backbone network. 
My question is how to visualize every layer after we give an image as input?
In other words, how can I see what my input image looks like in each layer in ResNET  of Detectron2?
(I mean save p2, p3, p4, p5 out as image file.)
Thanks a lot for your help!

**Comment by sushmasuresh28** [comment]
@107061218,  I am trying to do the exact same thing and can't seem to find sufficient documentation anywhere to get started.
Would be extremely grateful for any help or direction in this regard!

**Comment by lyhsieh** [comment]
I'm also desperately waiting for help XD.
What I want to do is just save p2, p3, p4, p5 out as image file.
Does anyone have good ideas?

**Comment by BWbwchen** [comment, marked as answer]
Hi, I solve this problem with @107061218 

For p2, p3, p4, p5, p6, they are in the FPN. 
So just go to [here](https://github.com/facebookresearch/detectron2/blob/0954ef32ef850e6435e3f5c2051c19f50ccb74db/detectron2/modeling/backbone/fpn.py#L154), the return value is a dictionary with keys p2, p3, p4, p5, p6 and corresponding feature map.

**Reply by lyhsieh** [reply]
Oh wow!
It does work!
So fantastic, so fun!
Thank you @BWbwchen !
Merci beaucoup!
Danke!

**Reply by sushmasuresh28** [reply]
Hi @107061218 and @BWbwchen !

It's wonderful that you were able to make it work! I am kinda feeling lost at the moment. Could you please help me get started?

How can I obtain p2, p3, p4, p5..?

Been stuck on this for days. Would really appreciate the help!

**Reply by BWbwchen** [reply]
Hi @sushmasuresh28 ,

As the answer I answered above, you can find the dictionary with key p2, p3, p4, p5 and p6.
And then, you should modify the source code to get the value and use it.
